mllib/src/main/scala/org/apache/spark/mllib/clustering/PowerIterationClustering.scala

```diff
@@ -205,7 +207,10 @@ class PowerIterationClustering private[clustering] (
       case "random" => randomInit(w)
       case "degree" => initDegreeVector(w)
     }
-    pic(w0)
+
+   // Materialized the graph w0 in randomInit/initDegreeVector, hence we can unpersist w.
```

Comment:
Oops, one more nit: this needs to indent 1 more space

Reply:
Thanks. done.